workers/unity/Packages/io.improbable.gdk.core/Utility/EntitySnapshot.cs

```diff
@@ -0,0 +1,137 @@
+using System.Collections.Generic;
+using Improbable.Worker.CInterop;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    ///     A snapshot of a SpatialOS entity.
+    /// </summary>
+    /// <remarks>
+    ///     This copies entity components from <see cref="Improbable.Worker.CInterop.Entity" /> for long term storage.
+    ///     This may only be a partial snapshot of an entity.
+    ///     The components present depend on the component filter used when making the entity query.
+    /// </remarks>
+    public struct EntitySnapshot
+    {
+        private readonly Dictionary<uint, ISpatialComponentSnapshot> components;
+
+        /// <summary>
+        ///     Get the SpatialOS component snapshot if present.
+        /// </summary>
+        /// <returns> The component snapshot, if it exists, or null otherwise.</returns>
+        /// <typeparam name="T">The component type.</typeparam>
+        public T? GetComponentSnapshot<T>() where T : struct, ISpatialComponentSnapshot
```

Comment:
how about a `TryGetComponentSnapshot<T>()`?

Reply:
Yeah why not

Reply:
Done 👍